tls/client-tls-nonblocking.c

```diff
@@ -108,38 +183,48 @@ int main(int argc, char** argv)
     /* Get the server IPv4 address from the command line call */
     if (inet_pton(AF_INET, argv[1], &servAddr.sin_addr) != 1) {
         fprintf(stderr, "ERROR: invalid Address\n");
-        return -1;
+        ret = -1;
+        goto exit;
     }
 
 
-
     /* Connect to the server */
     while (connect(sockfd, (struct sockaddr*) &servAddr, sizeof(servAddr))
            == -1) {
+        if (errno == EAGAIN || errno == EWOULDBLOCK) {
+                /* no error, just non-blocking. Carry on. */
+            continue;
+        } else if(errno == EINPROGRESS || errno == EALREADY)
+            break;
         /* just keep looping until a connection is made */
+        fprintf(stderr, "ERROR: failed to connect %d\n\n", errno);
+        ret = -1;
+        goto exit;
     }
 
-
-
     /* Create a WOLFSSL object */
     if ((ssl = wolfSSL_new(ctx)) == NULL) {
         fprintf(stderr, "ERROR: failed to create WOLFSSL object\n");
-        return -1;
+        ret = -1;
+        goto exit;
     }
 
     /* Attach wolfSSL to the socket */
-    wolfSSL_set_fd(ssl, sockfd);
-
+    if ((ret = wolfSSL_set_fd(ssl, sockfd)) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "ERROR: Failed to set the file descriptor\n");
+        goto exit;
+    }
     /* Connect to wolfSSL on the server side */
-    while (wolfSSL_connect(ssl) != SSL_SUCCESS) {
-        if (wolfSSL_want_read(ssl)) {
+    while (wolfSSL_connect(ssl) != WOLFSSL_SUCCESS) {
+        if (wolfSSL_want_read(ssl) || wolfSSL_want_write(ssl)) {
             /* no error, just non-blocking. Carry on. */
             printf("Waiting for connection...\n");
-            sleep(1); /* cut down on spam */
+            sleep(1); /* cut down on spam */  
```

Comment:
Please remove the print and sleep here. If you run these tests you will see a delay here.